### PR TITLE
FOGL-1189 - handle Ping route based on allowPing=>True config

### PR DESCRIPTION
**Pull request by amarendra-dianomic**
Test result:
`===================================================== 9 passed in 0.47 seconds =====================================================
`
Test suite result:
`============================================= 883 passed, 12 skipped in 42.10 seconds ==============================================
`
**Checklist:**
- [x] unit test modification

- [x] Works well on Raspbian with https + auth mandatory 

- [x] Works well  with https + auth optional

- [x] warning and error logs added

- [x]  proper HTTP status code added for raise

### CODE REVIEW

File: python/foglamp/services/core/api/common.py [hunk 1]
```diff
@@ -41,6 +43,15 @@
            curl -X GET http://localhost:8081/foglamp/ping
     """
 
+    try:
+        auth_token = request.token
+    except AttributeError:
+        cfg_mgr = ConfigurationManager(connect.get_storage())
+        category_item = await cfg_mgr.get_category_item('rest_api', 'allowPing')
+        allow_ping = True if category_item['value'] == 'true' else False
+        if request.is_auth_optional is False and allow_ping is False:
```
Comment (ashish-jabble) [inline python/foglamp/services/core/api/common.py]: `request.is_auth_optional is True` # auth is Optional

File: python/foglamp/services/core/api/common.py [hunk 2]
```diff
@@ -41,6 +43,15 @@
            curl -X GET http://localhost:8081/foglamp/ping
     """
 
+    try:
+        auth_token = request.token
+    except AttributeError:
+        cfg_mgr = ConfigurationManager(connect.get_storage())
+        category_item = await cfg_mgr.get_category_item('rest_api', 'allowPing')
+        allow_ping = True if category_item['value'] == 'true' else False
```
Comment (ashish-jabble) [inline python/foglamp/services/core/api/common.py]: Configuration manager value saved as it is for below scenarios:
curl -X PUT -d '{"value": "True"}' http://localhost:8081/foglamp/category/rest_api/allowPing
curl -X PUT -d '{"value": "TRUE"}' http://localhost:8081/foglamp/category/rest_api/allowPing

So we should handle `category_item['value'].lower()`

Comment (amarendra-dianomic) [reply python/foglamp/services/core/api/common.py]: done

**Comment by praveen-garg** [comment]
In case of optional => allowPing should not matter

**Comment by amarendra-dianomic** [comment]
@ashish-jabble changes done per feedback. Pl check.

**Pull request merged**